locking/locks.go

```diff
@@ -154,6 +157,17 @@ func (c *Client) UnlockFileById(id string, force bool) error {
 	if err := c.cache.RemoveById(id); err != nil {
 		return fmt.Errorf("Error caching unlock information: %v", err)
 	}
+	path := unlockRes.Lock.Path
+	gitRoot, err := git.RootDir()
+	if err != nil {
+		return err
+	}
+	absPath := filepath.Join(gitRoot, path)
```

Comment:
The locking API guarantees that paths returned from a spec-compliant server will be absolute with the root being the root directory of the git repository.

That all being said, I think this might still be a safe (and good) thing to do, _if_ we first:

- Checked to see whether or not the `unlockRes.Lock.Path` is absolute (i.e., starts with a "/").
- If so, use that as is.
- If not, make a call to `filepath.Join` with the first argument being the `gitRoot`.

I think that this can be implemented fairly cleanly as:

```go
var lockPath = unlockRes.Lock.Path
if !path.IsAbs(lockPath) {
        gitRoot, err := git.RootDir()
        if err != nil {
                return err
        }

        lockPath = filepath.Join(gitRoot, lockPath)
}

// ...
```